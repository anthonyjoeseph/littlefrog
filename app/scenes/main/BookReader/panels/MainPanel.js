// app/BookReader/InteractionPanel.js

import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  Text
} from 'react-native';

import Button from 'react-native-button'

class MainPanel extends Component{

  render(){
      return (
          <View style={
            [
              this.props.style,
              {
                backgroundColor: 'green',
                flexDirection: 'column',
                justifyContent: 'space-around',
                alignItems: 'center'
              }
            ]
          }>
            <Button
              containerStyle={styles.buttonContainer}
              style={styles.button}
              onPress={this.props.onPressHome}>H</Button>
            <Button
              containerStyle={styles.buttonContainer}
              style={styles.button}
              onPress={this.props.onPressCharacter}>C</Button>
            <Button
              containerStyle={styles.buttonContainer}
              style={styles.button}
              onPress={this.props.onPressLanguage}>L</Button>
            <Button
              containerStyle={styles.buttonContainer}
              style={styles.button}
              onPress={this.props.onPressText}>T</Button>
          </View>
      );
  }
}

const styles = StyleSheet.create({
  buttonContainer:{
    padding:5, height:45, overflow:'hidden', borderRadius:40, backgroundColor: 'white'
  },
  button:{
    flex:1, textAlign: 'center', backgroundColor: '#FFFFFF', color: 'brown', fontSize:30
  }
});

export default MainPanel;
