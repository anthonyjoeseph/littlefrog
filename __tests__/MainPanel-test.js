// __tests__/MainPanel-test.js
import 'react-native';
import React from 'react';
import MainPanel from '../app/scenes/main/BookReader/panels/MainPanel';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <MainPanel
      style={{
        zIndex: 2,
        width:50,
        backgroundColor:'transparent'
      }}
      onPressHome={ () => { } }
      onPressCharacter={ (newCharacter) => { } }
      onPressLanguage={ (newLanguage) => { } }
      onPressBookText={ (newLanguage) => { } }
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
