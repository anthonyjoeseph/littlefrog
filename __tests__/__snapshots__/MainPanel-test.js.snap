exports[`test renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "width": 50,
        "zIndex": 2,
      },
    ]
  }>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "white",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-around",
        "zIndex": 2,
      }
    }>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits="button"
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 5,
          "zIndex": 2,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007aff",
              "fontFamily": "HelveticaNeue-Medium",
              "fontSize": 17,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            null,
            Object {
              "backgroundColor": "#FFFFFF",
              "color": "brown",
              "fontSize": 30,
              "textAlign": "center",
              "zIndex": 2,
            },
            null,
          ]
        }>
        H
      </Text>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits="button"
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 5,
          "zIndex": 2,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007aff",
              "fontFamily": "HelveticaNeue-Medium",
              "fontSize": 17,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            null,
            Object {
              "backgroundColor": "#FFFFFF",
              "color": "brown",
              "fontSize": 30,
              "textAlign": "center",
              "zIndex": 2,
            },
            null,
          ]
        }>
        C
      </Text>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits="button"
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 5,
          "zIndex": 2,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007aff",
              "fontFamily": "HelveticaNeue-Medium",
              "fontSize": 17,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            null,
            Object {
              "backgroundColor": "#FFFFFF",
              "color": "brown",
              "fontSize": 30,
              "textAlign": "center",
              "zIndex": 2,
            },
            null,
          ]
        }>
        L
      </Text>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits="button"
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 5,
          "zIndex": 2,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007aff",
              "fontFamily": "HelveticaNeue-Medium",
              "fontSize": 17,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            null,
            Object {
              "backgroundColor": "#FFFFFF",
              "color": "brown",
              "fontSize": 30,
              "textAlign": "center",
              "zIndex": 2,
            },
            null,
          ]
        }>
        T
      </Text>
    </View>
  </View>
</View>
`;
